core/claims/claimAuthorizeKSignBabyJub_test.go

```diff
@@ -19,28 +20,88 @@ func TestClaimAuthorizeKSignBabyJub(t *testing.T) {
 	}
 	// Add input data to the test vector
 	if generateTest {
-		testgen.SetTestValue("privateKey", "28156abe7fe2fd433dc9df969286b96666489bac508612d0e16593e944c4f69f")
+		testgen.SetTestValue("0_privateKey", "28156abe7fe2fd433dc9df969286b96666489bac508612d0e16593e944c4f69f")
 	}
 	// Create new claim
 	var k babyjub.PrivateKey
-	hexK := testgen.GetTestValue("privateKey").(string)
+	hexK := testgen.GetTestValue("0_privateKey").(string)
 	if _, err := hex.Decode(k[:], []byte(hexK)); err != nil {
 		panic(err)
 	}
 	pk := k.Public()
 	c0 := NewClaimAuthorizeKSignBabyJub(pk)
+	assert.True(t, merkletree.CheckEntryInField(*c0.Entry()))
 	c0.Version = 1
 	e := c0.Entry()
 	// Check claim against test vector
-	checkClaim(e, t)
+	testgen.CheckTestValue(t, "0_HIndex", e.HIndex().Hex())
```

Comment:
Please, use `checkClaim` here

Reply:
This is not using `checkClaim` on purpose, as is a function that expects to have only `HIndex`, `HValue`, `dataString` only one time for each testvector file.
This happens only in very simple and incomplete tests, but when we have more than 1 test for a claim (for example in the ClaimAuthorizeKSignBabyJub), we need multiple `HIndex` and `HValue` for the multiple input output checks.
I thought about adding a parameter to the function `checkClaim` to specify which index of tests is using, but ended up viewing that is better to have directly the checks in the test code, having fater transparency on reading the test, instead of calling a function in another package.